bazel/genrule_repository.bzl

```diff
@@ -0,0 +1,49 @@
+def _genrule_repository(ctx):
+  ctx.download_and_extract(
+      ctx.attr.urls,
+      "", # output
+      ctx.attr.sha256,
+      "", # type
+      ctx.attr.strip_prefix,
+  )
+  for ii, patch in enumerate(ctx.attr.patches):
+    patch_input = "patch-input-%d.patch" % (ii,)
+    ctx.symlink(patch, patch_input)
+    patch_result = ctx.execute(["patch", "-p0", "--input", patch_input])
+    if patch_result.return_code != 0:
+      fail("Failed to apply patch %r: %s" % (patch, patch_result.stderr))
+
+  ctx.symlink(ctx.attr.genrule_cmd_file, "envoy-genrule-cmd.sh")
+  cat_genrule_cmd = ctx.execute(["cat", "envoy-genrule-cmd.sh"])
+  if cat_genrule_cmd.return_code != 0:
+    fail("Failed to read genrule command %r: %s" % (
+        ctx.attr.genrule_cmd_file, cat_genrule_cmd.stderr))
+
+  ctx.file("WORKSPACE", "workspace(name=%r)" % (ctx.name,))
+  ctx.template("BUILD.bazel", ctx.attr.build_file, {
+      "GENRULE_REPOSITORY_COMMAND": repr(cat_genrule_cmd.stdout.strip()),
+  })
```

Comment:
Prefer 4 spaces here. We've been a bit inconsistent, but the idea is that this Skylark, not Python and we're trying to be consistent across BUILD files and .bzl.

Reply:
Done. Also ran buildifier.